sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/implementation/KnowledgeAgentsImpl.java

```diff
@@ -0,0 +1,811 @@
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package com.azure.search.documents.indexes.implementation;
+
+import com.azure.core.annotation.BodyParam;
+import com.azure.core.annotation.Delete;
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.Get;
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.PathParam;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.Put;
+import com.azure.core.annotation.QueryParam;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.search.documents.indexes.implementation.models.ErrorResponseException;
+import com.azure.search.documents.indexes.implementation.models.RequestOptions;
+import com.azure.search.documents.indexes.models.KnowledgeAgent;
+import com.azure.search.documents.indexes.models.ListKnowledgeAgentsResult;
+import java.util.UUID;
+import reactor.core.publisher.Mono;
+
+/**
+ * An instance of this class provides access to all the operations defined in KnowledgeAgents.
+ */
+public final class KnowledgeAgentsImpl {
+    /**
+     * The proxy service used to perform REST calls.
+     */
+    private final KnowledgeAgentsService service;
+
+    /**
+     * The service client containing this operation class.
+     */
+    private final SearchServiceClientImpl client;
+
+    /**
+     * Initializes an instance of KnowledgeAgentsImpl.
+     * 
+     * @param client the instance of the service client containing this operation class.
+     */
+    KnowledgeAgentsImpl(SearchServiceClientImpl client) {
+        this.service
+            = RestProxy.create(KnowledgeAgentsService.class, client.getHttpPipeline(), client.getSerializerAdapter());
+        this.client = client;
+    }
+
+    /**
+     * The interface defining all the services for SearchServiceClientKnowledgeAgents to be used by the proxy service to
+     * perform REST calls.
+     */
+    @Host("{endpoint}")
+    @ServiceInterface(name = "SearchServiceClientK")
+    public interface KnowledgeAgentsService {
+        @Put("/agents('{agentName}')")
+        @ExpectedResponses({ 200, 201 })
+        @UnexpectedResponseExceptionType(ErrorResponseException.class)
+        Mono<Response<KnowledgeAgent>> createOrUpdate(@HostParam("endpoint") String endpoint,
```

Comment:
The APIs in this generated interface need to be exposed through the `SearchIndexClient`